lib/Runtime/Executor/ThreadPoolExecutor.cpp

```diff
@@ -0,0 +1,339 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ThreadPoolExecutor.h"
+
+#include <queue>
+#include <unordered_set>
+
+namespace glow {
+namespace runtime {
+
+ExecutionState::ExecutionState(RunIdentifierTy id, const DAGNode *root,
+                               DoneCbTy doneCb)
+    : cb_(doneCb), resultCtx_(std::make_unique<Context>()), inflightNodes_(0),
+      resultCode_(ResultCode::Executed) {
```

Comment:
Yeah, this is set up before a run. My idea was to assume that execution is going well until we find out otherwise (when a thread sets it to `Failed`).